go-controller/pkg/libovsdbops/db_index.go

```diff
@@ -0,0 +1,360 @@
+package libovsdbops
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+const (
+	// db ownership ExternalIDs, shared by all controllers
+	// OwnerController and OwnerObjectType define managing entity (a.k.a. owner) for given db object
+	OwnerController = types.OvnK8sPrefix + "/controller"
+	OwnerObjectType = types.OvnK8sPrefix + "/object-type"
+	// SecondaryIndexKey is a part of a secondary index, together with OwnerController and OwnerObjectType
+	// May be used by controllers to store e.g. namespace+name of the object.
+	SecondaryIndexKey = types.OvnK8sPrefix + "/name"
+	// PrimaryIDKey will be used as a primary index, that is unique for every db object,
+	// and can be built based on the combination of all the other ids.
+	PrimaryIDKey = types.OvnK8sPrefix + "/id"
+)
+
+type dbObjType int
+
+const (
+	addressSet dbObjType = iota
+	// ACL here
+)
+
+type controllerType int
+
+const (
+	defaultController controllerType = iota
+	// secondaryLayer3Controller here
+)
+
+type ownerType string
+
+const (
+	// owner types
+	EgressFirewallDNSOwnerType ownerType = "EgressFirewallDNS"
+	EgressQoSOwnerType         ownerType = "EgressQoS"
+	NetworkPolicyOwnerType     ownerType = "NetworkPolicy"
+	NamespaceOwnerType         ownerType = "Namespace"
+	HybridNodeRouteOwnerType   ownerType = "HybridNodeRoute"
+	EgressIPOwnerType          ownerType = "EgressIP"
+	EgressServiceOwnerType     ownerType = "EgressService"
+
+	// owner extra IDs
+	EgressQosAddrSetPriorityKey = "priority"
+	PolicyDirectionID           = "direction"
+	GressIdxID                  = "gress-index"
+	AddressSetIPFamilyID        = "ip-family"
```

Comment:
nit: could we have all these ending in `Key` for commonality and clarity of what they are.